extensions/amp-mustache/0.1/amp-mustache.js

```diff
@@ -60,23 +60,43 @@ export class AmpMustache extends AMP.BaseTemplate {
     }
     /** @private @const {!JsonObject} */
     this.nestedTemplates_ = dict();
-    let index = 0;
-    const content = templateContentClone(this.element);
-    iterateCursor(content.querySelectorAll('template'), nestedTemplate => {
-      const nestedTemplateKey = `__AMP_NESTED_TEMPLATE_${index}`;
-      this.nestedTemplates_[nestedTemplateKey] = nestedTemplate./*OK*/outerHTML;
 
+    /** @private {string} */
+    this.template_ = '';
+    if (this.element.tagName == 'TEMPLATE') {
+      const content = templateContentClone(this.element);
+      this.processNestedTemplates_(content);
+      const container = this.element.ownerDocument.createElement('div');
+      container.appendChild(content);
+      this.template_ = container./*OK*/innerHTML;
+    } else if (this.element.tagName == 'SCRIPT') {
+      //TODO(alabiaga): handle nested templates.
+      this.template_ = this.element.text.trim();
```

Comment:
We don't trim in the `<template>` case.

Reply:
right. mustache parsing won't care about the spaces anyway. removed in the script case.

Reply:
right. removed. mustache parse will ignore anyway